framework/Optimizers/OptimizerBase.py

```diff
@@ -0,0 +1,515 @@
+      @ In, denorm, bool, optional, if True the input data will be denormalized before returning
+      @ Out, prefix, str, #_#_#
+      @ Out, point, dict, {var:val}
+    """
+    try:
+      entry = self.submissionQueue[traj].popleft()
+    except IndexError:
+      self.raiseAnError(RuntimeError,'Tried to get a point from submission queue of trajectory "{}" but it is empty!'.format(traj))
+    prefix = entry['prefix']
+    point = entry['inputs']
+    if denorm:
+      point = self.denormalizeData(point)
+    return prefix,point
+
+  def getInitParams(self):
+    """
+      This function is called from the base class to print some of the information inside the class.
+      Whatever is permanent in the class and not inherited from the parent class should be mentioned here.
+      The information is passed back in the dictionary. No information about values that change during the simulation
+      are allowed
+      @ In, None
+      @ Out, paramDict, dict, dictionary containing the parameter names as keys and each parameter's initial value as
+        the dictionary values
+    """
+    paramDict = {}
+    for variable in self.getOptVars(full=True):
+      paramDict[variable] = 'is sampled as a decision variable'
+    paramDict['limit_mdlEval' ]        = self.limit['mdlEval']
+    paramDict['limit_optIter']         = self.limit['varsUpdate']
+    paramDict.update(self.localGetInitParams())
+    return paramDict
+
+  def getCurrentSetting(self):
+    """
+      This function is called from the base class to print some of the information inside the class.
+      Whatever is a temporary value in the class and not inherited from the parent class should be mentioned here
+      The information is passed back in the dictionary
+      @ In, None
+      @ Out, paramDict, dict, dictionary containing the parameter names as keys
+                              and each parameter's initial value as the dictionary values
+    """
+    paramDict = Sampler.getCurrentSetting(self)
+    paramDict.pop('counter', None)
+    paramDict['counter_mdlEval'       ] = self.counter['mdlEval']
+    paramDict['counter_varsUpdate'    ] = self.counter['varsUpdate']
+    paramDict.update(self.localGetCurrentSetting())
+    return paramDict
+
+  def initialize(self,externalSeeding=None,solutionExport=None):
+    """
+      This function should be called every time a clean optimizer is needed. Called before takeAstep in <Step>
+      @ In, externalSeeding, int, optional, external seed
+      @ In, solutionExport, DataObject, optional, a PointSet to hold the solution
+      @ Out, None
+    """
+    # NOTE: counter['varsUpdate'] needs to be set AFTER self.optTraj length is set by the sampler (if used exclusively)
+    self.counter['mdlEval'] = 0
+    self.counter['varsUpdate'] = [0]*len(self.optTraj)
+    self.optTrajLive = copy.deepcopy(self.optTraj)
+    # TODO: We should use retrieveObjectFromAssemblerDict to get the Instance
```

Comment:
Is there a Sampler method we should be leveraging here? Seeing the "[0][3]" makes me thing there's probably room for a method to separate this one level from the Assembler mechanics.